### ISSUE TITLE
Pretty print test output

### ISSUE THREAD

**Issue by Neppord** [tree-sitter/tree-sitter#655]
When receiving a mismatch between the expected and actual AST  it is not formatted in a readable way. instead it is all on one line.

It would be nice if it was printed in a more readable format. I try to use the tests as a feed backloop while developing the parser and they are just not readable enough for that.

**Comment by milahu** [comment]
another downside of the one-line-color-coded diff format is:
you cannot copy-paste that to markdown
instead, you need a screenshot image, to copy the colors

suggestion: annotated diff format
```diff
+++ expected
--- actual
 (fragment
-  (ERROR (tag_name))          ; <script/>
   (script_element
+    (start_tag (tag_name))    ; <script/>
-    (start_tag (tag_name))    ; <script>
+    (text)                    ; \n<script>true
-    (text)                    ; true
     (end_tag (tag_name))      ; </script>
   )
 )
```
([source](https://github.com/tree-sitter/tree-sitter-html/pull/18))

the grouping of nodes to lines would be suggested by the corpus/spec.txt files

indent would be ignored to reduce diffs (?)

affected function
https://github.com/tree-sitter/tree-sitter/blob/53949b09fdca24bb9d3c17df0404fa96c6ced2f1/cli/src/test.rs#L127

problem: diff files cannot be indented
currently, the diff-line is indented with four spaces

<details>

```diff
    +++ expected
    --- actual
     (fragment
    -  (ERROR (tag_name))          ; <script/>
       (script_element
    +    (start_tag (tag_name))    ; <script/>
    -    (start_tag (tag_name))    ; <script>
    +    (text)                    ; \n<script>true
    -    (text)                    ; true
         (end_tag (tag_name))      ; </script>
       )
     )
```

</details>

here is a start

<details>

```rust
pub fn print_diff(actual: &String, expected: &String) {
    println!(
        "{}\n{}",
        Colour::Green.paint("+++ expected"),
        Colour::Red.paint("--- actual")
    );
    let changeset = Changeset::new(actual, expected, " ");
    // TODO auto-indent nodes
    // TODO annotate source strings
    for diff in &changeset.diffs {
        match diff {
            Difference::Same(part) => {
                println!("  {}", part);
            }
            Difference::Add(part) => {
                println!("{}", Colour::Green.paint(format!("+ {}", part)));
            }
            Difference::Rem(part) => {
                println!("{}", Colour::Red.paint(format!("- {}", part)));
            }
        }
    }
    println!("");
}
```

result

```diff
+++ expected
--- actual
- (fragment (ERROR (tag_name)) (text)
+ (fragment
  (script_element (start_tag (tag_name)) (raw_text) (end_tag (tag_name))))

```
"(fragment" should be same
looks like [a bug](https://github.com/johannhof/difference.rs/issues/47) in difference::Changeset (too greedy/lazy)

</details>

**Comment by ahlinc** [comment]
The case in description seems was fixed in #1278

**Comment by ahelwer** [comment]
One thing it would be helpful to address as a special case is the number of trailing right-parentheses. If I have the wrong number then it highlights the entire last node as an error case, rather than just showing it expected a few more or fewer parens.
